lesson3.js

```diff
@@ -0,0 +1,48 @@
+// задание 1
+var i = 0;
+sqrtPoint = [];
+point:
+    while (i++ <= 5000) {
+        if ((i % 2 === 0 && i !== 2) || i === 0 || i === 1) {  // Отметаем вывод четных чисел, нуля и единицы
+            continue;
+        }
+        if (Math.sqrt(i) % 1 === 0) {                          // Отметает вывод чисел имеющих корень с добавлением
+            sqrtPoint.push(Math.sqrt(i));                      // корней в массив, что бы в дальнейщем уменьшить
+            continue;                                          // к-во проверок на возможные делители
+        } else {
+            for (j = 0; j <= sqrtPoint.length; j++) {          // Проверяем числа на деление используя корни из массива
+                if (i % sqrtPoint[j] === 0) {
+                    continue point;
+                }
+            }
+        }
+        console.log(i);
+    }
+/*
+    Захотелось самому написать алгоритм, работает с любым числовым промежутком,
+    но я знаю, что можно просто сделать цикл в цикле или применить решето.
+*/
+
+
+// Задание 2
+var cart = [["prodName", "color", "size", 150], ["prodName", "color", "size", 150], ["prodName", "color", "size", 150]];
+function countBasketPrice() {
+    var sumPrice = 0;
+    for (i = 0; i < cart.length; i++) {
+        sumPrice += cart[i][3];
+    }
+    return sumPrice;
+}
+alert(countBasketPrice());
+
+
+// Задание 3
+    for (i = 0; i <= 9; console.log(i++)){}
+
+
+// Задание 4
+x = "x";
```

Comment:
Использовать переменные без var - плохая практика. На последнем уроке нашего курса рассмотрим почему
```suggestion
var x = "x";
```